mvg/analysis_classes.py

```diff
@@ -188,18 +254,32 @@ def summary(self):
         self.check_status()
         # print time info if applicable
         if self.time_column is not None:
-            from_t = self.dframe[self.time_column].min()
-            to_t = self.dframe[self.time_column].max()
+            from_t = self._dframe[self.time_column].min()
+            to_t = self._dframe[self.time_column].max()
             if self.time_column == "datetime":
                 from_t = from_t.strftime("%Y%m%d-%H:%M.%S")
                 to_t = to_t.strftime("%Y%m%d-%H:%M.%S")
             print(f"from {from_t} to {to_t}")
 
     # Default method
-    def plot(self):
-        """ Pro forma ancestor function"""
+    def plot(self, interactive=True):  # pylint: disable=unused-argument
+        """Pro forma ancestor function.
+
+        Parameters
+        ----------
+        interactive : bool
+            True: show plot, False: save plot
+
+
+        Returns
+        -------
+        plot file name: str
+          name of plot file (or emtpy string in case of interactive plot)
+        """
+
         self.check_status()
         print(f"Plot function not implemented for {type(self).__name__}")
+        return ""
```

Comment:
Maybe `raise NotImplementedError` or make abstract method. I do not see a situation where you would want to `super()` this method, only overwriting it.

Reply:
The design idea is that all analysis shall provide all methods implemented in Analysis. The intended case is an analysis where it does not make sense to plot something (we have no case right now...) and instead of raising an error there will just be a text message.